### ISSUE TITLE
PersistentVolumeClaim cannot be deleted.

### ISSUE THREAD

**Issue by Black-max12138** [kubernetes/kubernetes#128695]
### What happened?

https://github.com/kubernetes/kubernetes/blob/c25f5eefe4efda4c0d9561d06942cd3de3dfe2e4/pkg/controller/volume/pvcprotection/pvc_protection_controller.go#L374-L388
If a pod with UnexpectedAdmissionError exists in the environment and PersistentVolumeClaim is used, the PVC cannot be deleted after the pod is deleted. The controller-manager log shows that Pod uses PVC xxx. Check the code and find that the podUsesPVC method does not determine the pod status. I think this is a problem.

### What did you expect to happen?

The pvc should be deleted correctly.

### How can we reproduce it (as minimally and precisely as possible)?

Construct a pod in the UnexpectedAdmissionError state, configure a PVC, and delete the pod that is using the PVC.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```
1.31
</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by JayantDhakajd** [comment]
Good afternoon, @Black-max12138 ! Would it be okay to add a check for unexpectedAdmissionError by implementing a function in the code? If you agree, I can also test the changes using a test file to ensure everything works as expected. 
Thanks!

**Comment by Black-max12138** [comment]
> Good afternoon, @Black-max12138 ! Would it be okay to add a check for unexpectedAdmissionError by implementing a function in the code? If you agree, I can also test the changes using a test file to ensure everything works as expected. Thanks!

Yes, this problem can be solved if the pod status judgment is added to the preceding code.

**Comment by JayantDhakajd** [comment]
Good Morning, @Black-max12138 I had made some changes in the code and added a function to look for UnexpectedAdmissionError, but I'm confused on how to test it by creating a pod with UnexpectedAdmissionError, If you can suggest me what to do after or should i create a pull request just just changing the code ..? 
Thanks !

**Comment by JayantDhakajd** [comment]
\assign

**Comment by niranjandarshann** [comment]
The issue is that podUsesPVC does not check for the pod's status before determining whether the PVC is still in use. If a pod is in the UnexpectedAdmissionError state, it should not block PVC deletion. so if we add the pod status check at the begining of the function then i think it will  Skip pods in an error state to allow PVC deletion.

**Comment by niranjandarshann** [comment]
@JayantDhakajd  are you working on this issue?
